corecursive equations for M-types

dominik-kirst opened the pull request:
I've adopted Felix's idea to work with Brandon's construction, establishing definitional corecursion. However, it's not completely finished as the proof that the hprop one factors through is actually an hprop is not done yet. Also there's a functoriality equality missing which should hold trivially by finality but I couldn't find the right way to prove it.

langston-barrett committed:
hfiber as a pullback of diagrams involving arrows from unit

langston-barrett committed:
Remove extraneous lemma transitive_paths_weq_reverse

langston-barrett committed:
Merge remote-tracking branch 'origin/master' into pb-inverse-image

langston-barrett committed:
Merge branch 'master' into pb-inverse-image

langston-barrett committed:
Merge branch 'master' into pb-inverse-image

langston-barrett committed:
Merge remote-tracking branch 'origin/master' into pb-inverse-image

langston-barrett committed:
Make proof non-opaque

vbgl committed:
Declare hint databases in the prelude

langston-barrett committed:
Reorganize material on the precategory of types

maggesi committed:
Remove simple_pair_path, duplicate of total2_paths2.

DanGrayson committed:
Merge pull request #1193 from maggesi/rm-simple_pair_path

Remove simple_pair_path, duplicate of total2_paths2.

Commit:
Material from UniMath school 2019: Non-wellfounded trees

Coauthored with Christian Graulund

langston-barrett committed:
Merge branch 'master' into types

benediktahrens committed:
Merge pull request #1194 from vbgl/rm-hardwired-hint-db

Declare hint databases in the prelude

langston-barrett committed:
Merge branch 'master' into types

langston-barrett committed:
reflective subcategories

langston-barrett committed:
reflective subcategories

dominik-kirst committed:
corecursive equation

dominik-kirst committed:
morphism equation and refactoring

vbgl committed:
Minor cleaning

benediktahrens committed:
Merge pull request #1196 from vbgl/useless-rewrite

Minor cleaning

DanGrayson committed:
fix Makefile so test files aren't include in All.v files

benediktahrens committed:
Merge pull request #1197 from DanGrayson/no-test-files-in-all-files

fix Makefile so test files aren't include in All.v files

esg committed:
Fix a misleading comment.

benediktahrens committed:
Merge pull request #1200 from esg/comment-fix

Fix a misleading comment.

dominik-kirst committed:
carriers are equal

dominik-kirst committed:
still working on equations

rmatthes commented:
a shallow comment: please do not use "reflexivity" that is more complex in Coq than a simple application of idpath (which is notation for the constructor paths_refl)

mroman42 committed:
Add swapping functor

This functor is useful when defining the braided structure of monoidal
categories and endows Cat with braided monoidal structure.

mroman42 committed:
Add braided monoidal precategories

Defines braidings and the braided monoidal structure on top of them.
They are required to satisfy the hexagon equations. Accessors and
a coercion from braided monoidal to monoidal are provided.

In #722, braided monoidal categories were discussed, but the pull
request was too big to be merged. In #1009, monoidal categories were
added, but braided monoidal categories were not part of the pull
request.

Note that this adds only braided monoidal PREcategories, even when
the name of the file tries to match MonoidalCategories.v, which only
provides precategories.

Also update .package/files.

mroman42 committed:
Monoidal categories as bicategories

Every monoidal category gives rise to a bicategory where the type of
0-cells is the unit type.  We check that all the equations on the
definition of a prebicategory are satisfied by the monoidal precategory.

The only difficult one is the pentagon equation, that reads backwards
on one of the two definitions.

It should be easy to extend this to monoidal categories and
bicategories.  The hope would be that we can use some of the general results
from bicategories in monoidal categories without having to reprove
them again.

dominik-kirst committed:
proved finality plus cleanup

rmatthes committed:
inform makefile about ComputationalM.v, replace reflexivity by apply idpath

mroman42 committed:
Update .package/files

benediktahrens committed:
Merge pull request #1201 from mroman42/braided-monoidal

Braided monoidal precategories

langston-barrett committed:
Merge branch 'master' into types

langston-barrett committed:
Counit of an adjunction is epi if the right adjoint is faithful

langston-barrett committed:
Counit of an adjunction is epi if the right adjoint is faithful

JLimperg committed:
Injective pairing is inconsistent with univalence

Injective pairing is the proposition

  (a ,, b) = (a ,, b') -> b = b'.

This commit adds a proof that injective pairing is inconsistent with
univalence.

JLimperg committed:
Injective pairing is inconsistent with univalence

Injective pairing is the proposition

  (a ,, b) = (a ,, b') -> b = b'.

This commit adds a proof that injective pairing is inconsistent with
univalence.

JLimperg committed:
Injective pairing is equivalent to UIP

This commit adds a proof that injective pairing is equivalent to UIP
(aka K), which makes it clear that it can't be consistent with
univalence.

JLimperg committed:
Injective pairing is equivalent to UIP

This commit adds a proof that injective pairing is equivalent to UIP
(aka K), which makes it clear that it can't be consistent with
univalence.

langston-barrett committed:
Split monomorphisms

langston-barrett committed:
Split monomorphisms

langston-barrett committed:
Put split monics and epis in their own file

Also prove that an epic split monic is an iso

langston-barrett committed:
Put split monics and epis in their own file

Also prove that an epic split monic is an iso

langston-barrett committed:
The localization of a reflective subcategory is idempotent

langston-barrett committed:
The localization of a reflective subcategory is idempotent

langston-barrett committed:
Merge branch 'master' of ssh://github.com/UniMath/UniMath into pb-inverse-image

langston-barrett committed:
Remove accidental commits on Makefile

langston-barrett committed:
Remove extraneous file

langston-barrett committed:
Remove extraneous material

langston-barrett committed:
Remove extraneous material

langston-barrett committed:
Add result on simplifying cochain cones

benediktahrens committed:
Merge branch 'master' into monoidal-is-bicategory

benediktahrens committed:
Merge pull request #1132 from siddharthist/types

Reorganize results about the precategory of types

benediktahrens committed:
Merge branch 'master' into pb-inverse-image

mroman42 committed:
Use '?' instead of introducing unused identifiers

Sometimes even 'intros' is enough and the names of the unused identifiers
were not very informative.

mroman42 committed:
Merge branch 'master' into monoidal-is-bicategory

DanGrayson committed:
Merge pull request #1090 from siddharthist/pb-inverse-image

hfiber as a pullback of arrows from unit

benediktahrens committed:
Merge branch 'master' into monoidal-is-bicategory

benediktahrens committed:
Merge pull request #1202 from mroman42/monoidal-is-bicategory

Monoidal precategories as prebicategories

langston-barrett committed:
Merge remote-tracking branch 'upstream/master' into adjunction-lemma

langston-barrett committed:
Fix changed name (EpisAreSplit -> epis_are_split), CONTENTS

langston-barrett committed:
Fix changed name (EpisAreSplit -> epis_are_split), CONTENTS

JLimperg committed:
NoInjectivePairing: Fix misleading comment

JLimperg committed:
NoInjectivePairing: Fix misleading comment

JLimperg committed:
NoInjectivePairing: Nicer proof of pathsinv0_weqtopaths

JLimperg committed:
NoInjectivePairing: Nicer proof of pathsinv0_weqtopaths

rmatthes committed:
simple style changes

- no use of reflexivity, even less of replace that introduces Coq equality
- simpl -> cbn
- shorter lines
- more bullets
- comment on deviation from Mac Lane

rmatthes committed:
simple style changes

- no reflexivity
- unfold -> red
- simpl -> cbn
- shorter lines

rmatthes committed:
mostly style changes, but also guarantee homsets of that category

- unfold -> red
- simpl -> cbn
- identify is_monoid_mor as proposition (avoid repetition of that argument)
- comment on deviation from Mac Lane
- homsets are carried over from the base category

rmatthes committed:
style changes

- simpl -> cbn
- unfold -> red
- more bullets
- apply -> exact when possible

rmatthes committed:
style changes but also more modular definitions, needed small upstream development

- upstream: UniMath/CategoryTheory/PrecategoryBinProduct.v now contains lemmas to give a nat. transformation when plugging a morphism as fixed argument into a functor on a product category
- more bullets
- simpl -> cbn
- less peculiar way to change the type of hypotheses or the goal
- the functors and nat. tranformations had been mostly defined from scratch instead of composing elements from the library; now much less so, which shortens the code considerably
- shorten lines through line breaks

rmatthes committed:
minor work on whitespace + avoid reflexivity

DanGrayson committed:
change name of "development team"

benediktahrens committed:
Merge pull request #1209 from DanGrayson/change-committee-name-2

change name of "development team"

rmatthes committed:
continue the process of streamlining definitions

- replace two more "manual" functor definitions by compact definitions involving constructions from the library
- add "ok" lemmas for all the replaced definitions spelling out explicitly the very first component of the functor/nat. transformation that had been given in the original "manual" construction - not meant to be used since they fall within convertibility

rmatthes committed:
execute the streamlining on the 4 functors of this file

nmvdw committed:
Chaotic displayed bicats

nmvdw committed:
Merge branch 'master' into chaotic

rmatthes committed:
get rid of longest proof of naturality of a hand-written definition

- identify the construction of parameterized post-whiskering upstream (in UniMath/CategoryTheory/PrecategoryBinProduct.v)
- use several whiskerings to define the natural transformation U_action_χ_nat_trans
- give a lemma to show what that transformation does on elements

rmatthes committed:
apply the method to the two "manual" functor definitions of this file

rmatthes committed:
Merge branch 'master' into somemodifonmonoidal

rmatthes committed:
added comments to the file header + more use of symbol for natural transformation

benediktahrens committed:
Merge pull request #1210 from nmvdw/chaotic

Property which says when a displayed bicat is chaotic

rmatthes committed:
Merge branch 'master' into somemodifonmonoidal

benediktahrens committed:
Merge pull request #1208 from rmatthes/somemodifonmonoidal

working on the package on monoidal categories

rmatthes committed:
θ_source and θ_target in Signatures.v now defined by composition from library constructions

everything else is either a very light cleaning or necessary consequence of this change

rmatthes committed:
a little bit of beautification

langston-barrett committed:
Define the syntax of propositional logic

langston-barrett committed:
Define the syntax of propositional logic

langston-barrett committed:
Propositional logic: define iff

langston-barrett committed:
Propositional logic: define iff

rmatthes committed:
added results on swapping of the arguments of the tensor in monoidal categories

- monoidal category stays monoidal when swapping, and monoidal functors stay monoidal when tensors in source and target are each turned around
- proofs were rather cumbersome
- a small construction was added upstream (PrecategoryBinProduct.v), but also some beautification to that file was done

JLimperg committed:
NoInjectivePairing: Remove redundant lemma

JLimperg committed:
NoInjectivePairing: Remove redundant lemma

JLimperg committed:
NoInjectivePairing: Avoid evars in negb_weq_pair

JLimperg committed:
NoInjectivePairing: Avoid evars in negb_weq_pair

JLimperg committed:
NoInjectivePairing: Avoid rewrite in pathsinv0_weqtopaths

JLimperg committed:
NoInjectivePairing: Avoid rewrite in pathsinv0_weqtopaths

benediktahrens committed:
Merge branch 'master' into no-injective-pairing

langston-barrett committed:
HSET -> hSet (remove explicit pr1)

langston-barrett committed:
HSET -> hSet (remove explicit pr1)

langston-barrett committed:
Create MoreFoundations/Bool.v

langston-barrett committed:
Create MoreFoundations/Bool.v

langston-barrett committed:
Merge remote-tracking branch 'upstream/master' into propositional-logic

langston-barrett committed:
Remove outdated TODO

langston-barrett committed:
Remove outdated TODO

langston-barrett committed:
Merge remote-tracking branch 'upstream/master' into m2

langston-barrett committed:
Upstream some lemmas to MoreFoundations

langston-barrett committed:
Qed -> Defined

langston-barrett committed:
Merge remote-tracking branch 'upstream/master' into adjunction-lemma

langston-barrett committed:
List Subcategory/Reflective.v

langston-barrett committed:
List Subcategory/Reflective.v

benediktahrens committed:
Merge pull request #1204 from JLimperg/no-injective-pairing

Injective pairing is inconsistent with univalence

langston-barrett committed:
Merge branch 'master' into adjunction-lemma

DanGrayson committed:
uniform use of underscores

refer to #687

DanGrayson committed:
uniform use of underscores

refer to #687

rmatthes committed:
made the type of the functor for the strength more liberal

this is in Signatures.v, even if there is not yet a use of the new freedom;
it serves as a preparation for a generalization on the abstract level

all the other files had to be adapted

maggesi committed:
Uniform use of prefix 'make_' in builders names.

langston-barrett committed:
Move tests on Bool.v to their own file

langston-barrett committed:
Move tests on Bool.v to their own file

langston-barrett committed:
Merge remote-tracking branch 'upstream/master' into propositional-logic

langston-barrett committed:
Begin proof of induction principle for propositional logic

langston-barrett committed:
Begin proof of induction principle for propositional logic

benediktahrens committed:
Merge pull request #1215 from DanGrayson/choosing-identifiers

uniform use of underscores

benediktahrens committed:
Merge pull request #1205 from siddharthist/adjunction-lemma

Split monics/epis, (co)reflective subcategories, adjunction lemmas

benediktahrens committed:
Merge pull request #1211 from siddharthist/propositional-logic

Define the syntax of propositional logic

maggesi committed:
Merge branch 'master' into uniform_prefix_for_builders

rmatthes committed:
added a construction of monoidal categories from bicategories by fixing an object and taking its endomorphisms

problem: final "Defined" takes overly long - I did not see it come to an end

maggesi committed:
Fixes after merge

rmatthes committed:
modularize better the constructions

rmatthes committed:
further efforts for the reduction of compilation time (and modularization)

- variant of rassociator_transf in Bicat.v with different argument order
- an extra lemma is_iso_rassociator in Bicat.v for ease of later proofs
- continue BicategoryFromMonoidal.v: all the excess compilation time sits in nat_trans_associator

rmatthes committed:
endofunctors turned into a monoidal category

- the previous file EndofunctorsMonoidal.v is now called UnitorsAndAssociatorsForEndofunctors.v
  since it provides only the very beginning
- the new material is placed into the subpackage Monoidal since it is more of interest there and not
  yet needed in the package SubstitutionSystems

rmatthes committed:
Merge branch 'master' into onsignaturesandmonoidalcats

rmatthes committed:
internally optimized opacity - still very slow

rmatthes committed:
made an attempt to define monoidal_precat_struct as intermediary structure

rmatthes committed:
The file that had been announce in commit c44683a but that was forgotten to add.

rmatthes committed:
problem solved by Anders Mörtberg: now the crucial proof of nat_trans_associator goes fast

also reestablished computational contents of the lemmas that prove is_iso
some cleanup

rmatthes committed:
removed the devices with "abstract" that were meant to hide the computationally irrelevant components of structures

rmatthes committed:
big effort to make irrelevant parts of structures opaque

- choice of the treated files may seem rather arbitrary - I followed my intuition (sometimes only Qed in place of Defined, several times auxiliary definitions and lemmas before the construction, more use of "abstract")
- in passing: also used Unicode symbols a bit more often
- side effect on ModulesFromSignatures.v that had to be repaired

rmatthes committed:
very much reduced compilation time for swapping of arguments

just by taking out proofs of properties into auxiliary lemmas

rmatthes committed:
massive acceleration of compilation time

by the usual taking into pieces and Qed and "abstract"

benediktahrens committed:
Merge pull request #1216 from maggesi/uniform_prefix_for_builders

Uniform use of prefix 'make_' in builders names.

benediktahrens committed:
Merge branch 'master' into onsignaturesandmonoidalcats

benediktahrens committed:
fix problems introduced during merge

rmatthes committed:
Merge pull request #7 from benediktahrens/onsignaturesandmonoidalcats

fix problems introduced during merge

benediktahrens committed:
Merge pull request #1219 from rmatthes/onsignaturesandmonoidalcats

material on (semantic) signatures and monoidal categories

niccoloveltri committed:
representables pseudofunctors, yoneda embedding and biinitiality

niccoloveltri committed:
representables pseudofunctors, yoneda embedding and biinitiality, revisited

niccoloveltri committed:
opaque subproofs of is_inv2cell_to_is_iso in Univalence.v

benediktahrens committed:
Merge pull request #1222 from niccoloveltri/master

representables pseudofunctors, yoneda embedding and biinitiality

rmatthes committed:
Merge branch 'master' into continuepr1206

rmatthes committed:
two shallow changes to solve incompatibilities with the evolved master

rmatthes committed:
shallow changes to satisfy the reviewer (weq results are defined) + minor cleanup

- in M: two times the equality proofs for weq_iso made abstract
- removes the comment that the universal property is missing
- newline at end of all files
- Univalence.v: more harmonious typing of the first few lemmas, removes a spurious argument, moves the very last lemma up to where it belongs

rmatthes committed:
declares two elements as local, as suggested by Benedikt

rmatthes committed:
replaces some projections by accessors (very shallow contribution)

rmatthes committed:
Merge branch 'master' into computationalM

rmatthes committed:
shallow changes to reestablish compilation

rmatthes committed:
removes a comment, as suggested by Langston

benediktahrens committed:
Merge pull request #1224 from rmatthes/continuepr1206

Continue PR1206

rmatthes committed:
wip on establishing isprop for the crucial property P

this is by transfer of the already formalized proof by Felix Rech

rmatthes committed:
fills the gaps in the development: P proven a proposition, fine-tuning for proper behaviour

maximedenes committed:
Make sure bound scopes are imported correctly.

Scopes used for arguments depend on the bound scopes (in the sense of
`Bind Scope`) that have been imported (as in `Import`). A Coq bug was
making some non-imported bound scopes visibles, and one file of UniMath
was relying on this bug. We fix it here.

This is in preparation for https://github.com/coq/coq/pull/10181, but
should be backward compatible (remains to be tested).

benediktahrens committed:
Merge pull request #1225 from maximedenes/fix-bound-scopes

Make sure bound scopes are imported correctly.

benediktahrens committed:
update to Coq 8.9.1

DanGrayson committed:
Merge pull request #1229 from benediktahrens/coq8.9.1

update to Coq 8.9.1

dominik-kirst committed:
epsilon

dominik-kirst committed:
delta

rmatthes committed:
Merge branch 'master' into computationalM

rmatthes committed:
corrects a left-over from the merge process

rmatthes committed:
some cleanup

- uses function composition more often
- upstream lemmas were mostly in the library, one moved to MoreFoundations/PartA.v
- extended header information about the UniMath schools that contributed to the present file

dominik-kirst committed:
erased remaining two upstream lemmas

rmatthes commented:
I think that the number of commits that have nothing to do with the new contribution here is overwhelming. I'll try to construct a new branch that is nicer in this respect.

rmatthes committed:
modifies list of authors after email exchange with Dominik

rmatthes committed:
leaves out material on which project work at UniMathSchool 2019 was based

this does not give a clean commit history, but at least, the overall diff for the PR under construction will be readable

rmatthes committed:
tiny changes to minimize number of modified files in PR

rmatthes committed:
description of PR integrated into introductory comments

rmatthes committed:
Defined replaces Qed everywhere, except for the lemma proved by idpath

rmatthes committed:
last Qed turned into Defined
